### ISSUE TITLE
How to choose identical embeddings and language models for query with Chroma?

### ISSUE THREAD

**Issue by erlebach** [jerryjliu/llama_index#7368]
### Question Validation

- [X] I have searched both the documentation and discord for an answer.

### Question

I am using Llama-index with Chroma to create a persistent database. I then read the database and perform a query using `index.query_engine.query(str)`. The storage is a `ChromaVectorStore`, and the index is a `VectorStoreIndex`. This set up implies that the embedding model and the language model used for the query should be the same. Otherwise, the embeddings generated from the query would not be consistent with the embeddings in the database. (I assume I understand this correctly)? Furthermore, the language models and embeddings are set in the `ServiceContext`. So my question is the following: how can ensure that the `embed_model` model and the language model `llm` are the same. I sometimes get confused as to the difference between the embedding model and the language model. Here is some code:

```
        self.embed_model = LangchainEmbedding(
            # 384 dim-embeddings
            HuggingFaceEmbeddings(
                model_name="sentence-transformers/all-MiniLM-L6-v2",
                cache_folder="./transformer_cache",  # local download
            ),
            embed_batch_size=10,
        )
        llm_predictor = LLMPredictor()
        self.service_context = ServiceContext.from_defaults(
            llm=OpenAI(temperature=0.0, model="gpt-3.5-turbo"),
            embed_model=self.embed_model,  # This is the model used for querying
        )
```

The main line is the `self.service_context`. When I run a query after reloading the model from disk, I sometimes get a message such as:

`chromadb.errors.InvalidDimensionException: Embedding dimension 1536 does not match collection dimensionality 384. `

I would like to experiment with a range of embed_models and LLM models with different embedding sizes. For some reason, embedding sizes are not stored as easy to access information with the different models. I am not sure why. Any insight that would help me would be greatly appreciated. Thanks.

**Comment by logan-markewich** [comment]
@erlebach actually, the LLM can change any time. The only thing that has to be consistent is the embed model.

So whatever embedding model you used to build the index, you should make sure you use the same when loading/querying.

That error message implies that openai embeddings are being used to query the DB, and there is a mismatch there. An easy way to work with the service context is to just set it as global.

```
from llama_index import ServiceContext, set_global_service_context
service_context = ServiceContext.from_defaults(embed_model=..., ....)

set_global_service_context(service_context)
```

**Comment by erlebach** [comment]
Thanks. That did the trick. I had included the wrong `ServiceContextInstance` in one of my load statements. 

I had searched the Issues, but obviously missed the one you referred to, similar to mine. Sorry about that. 

Finally, is there a list, in one convenient place, of all the available embedding and llm models that one can use with Llamaindex? I have read the documentation, and know about `OpenAIEmbeddings`, etc, but have not seen a list of model names. Perhaps I missed it. Thanks.
